Feat: Product 생성 API

ghpark7 opened the pull request:
변경사항
-----
-ProductRepository 재생성 및 연동
-Product 등록 API
-Product Create Request 및 Service 작성

개발하면서 발생한 문제점(에러, 어려움)
-----
-잘 작동하는지 확인하기 위한 과정에서 발생한 무수한 에러...

해결방안 또는 새롭게 알게된 점
-----
-도커로 mysql 연동해서 API 테스트 해보는 법
-H2가 잘 안될때 간단한 수정으로 mysql로 test해보는 법
-int형에 비해 INTEGER형이 가지는 장점(레퍼런스형)

스크린샷
-----
당부하고 싶은 말 또는 논의해봐야할 점
-----
(생략가능)

ghpark7 committed:
Feat: product 생성 API

변경사항
ProductRepository 재생성 및 연동
Product 등록 API
Product Create Request 및 Service

개발하면서 발생한 문제점(에러, 어려움)
무수한 에러....

해결방안 또는 새롭게 알게된 점
도커로 mysql 연동해서 API 테스트 해보기

스크린샷

당부하고 싶은 말 또는 논의해봐야할 점

(생략가능)

ghpark7 committed:
Feat: product 생성 API

변경사항
ProductRepository 재생성 및 연동
Product 등록 API
Product Create Request 및 Service

개발하면서 발생한 문제점(에러, 어려움)
무수한 에러....

해결방안 또는 새롭게 알게된 점
도커로 mysql 연동해서 API 테스트 해보기

스크린샷

당부하고 싶은 말 또는 논의해봐야할 점

(생략가능)

GGHDMS commented on src/main/java/potato/potatoAPIserver/global/config/SecurityConfig.java:
```diff
@@ -39,6 +39,7 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 //요청에 대한 권한 설정
                 .authorizeHttpRequests((authorizeRequests) -> authorizeRequests
                         .requestMatchers("/api/v1/auth/**").permitAll() //해당 리소스에 대한 모든 요청 허가
+                        .requestMatchers("/products").permitAll()
```
로그인 한 유저만 상품 등록 하게 해야 되지 않아?

sunwon12 replied:
그러고 보니까 로그인 안 한 유저는 엑세스 토큰 안 가지고 있잖아 
게시글 조회 같은 API는 어떻게 요청해?

GGHDMS commented on src/main/java/potato/potatoAPIserver/product/api/ProductApi.java:
```diff
@@ -0,0 +1,27 @@
+ * @since: 2023-09-21
+ */
+@RestController
+@RequestMapping("/products")
```
/api/v1/products 형태로 해주라

ghpark7 replied:
아 오키 고쳤어!

GGHDMS commented on src/main/java/potato/potatoAPIserver/product/service/dto/ProductCreateRequest.java:
```diff
@@ -0,0 +1,23 @@
+ * @since: 2023-10-08
+ */
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
```
`@NoArgsConstructor` Dto 에서는 필요가 없을 듯 한데 혹시 사용한 이유가 있어??

그리고 생성자 를 직접 만들었는데 모든 param 을 받을 거면 ` @AllArgsConstructor` 를 사용하는게 더 좋을 수도

ghpark7 replied:
오케이 반영했습니다!

GGHDMS commented on src/main/java/potato/potatoAPIserver/product/api/ProductApi.java:
```diff
@@ -0,0 +1,27 @@
+
+    @PostMapping
+    void postProduct(@RequestBody ProductCreateRequest request){
+        productService.createProduct(request);
```
Post 를 통해 생성했으니깐 201 Created 를 사용합니다

 return new ResponseEntity<>(HttpStatus.CREATED); 이런식으로 ?

sunwon12 commented on src/main/java/potato/potatoAPIserver/product/api/ProductApi.java:
```diff
@@ -0,0 +1,27 @@
+ * @since: 2023-09-21
+ */
+@RestController
+@RequestMapping("/products")
```
ex) "api/v1/products"로 통일하자

ghpark7 replied:
오키이! 알겠습니닷

sunwon12 commented on src/main/java/potato/potatoAPIserver/global/config/SecurityConfig.java:
```diff
@@ -39,6 +39,7 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 //요청에 대한 권한 설정
                 .authorizeHttpRequests((authorizeRequests) -> authorizeRequests
                         .requestMatchers("/api/v1/auth/**").permitAll() //해당 리소스에 대한 모든 요청 허가
+                        .requestMatchers("/products").permitAll()
```
.permitAll()은 생략 가능하더라고 
.requestMatchers("/api/v1/auth/**").permitAll() //해당 리소스에 대한 모든 요청 허가
                        .requestMatchers("/products").permitAll()
앞으로 요거는 생략하자!

ghpark7 replied:
아아 생략되는거였구나 확인!!

sunwon12 commented on src/main/java/potato/potatoAPIserver/product/service/dto/ProductCreateRequest.java:
```diff
@@ -0,0 +1,23 @@
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class ProductCreateRequest {
+    private String title;
```
제목, 내용, (설명)은 필수 값이니까 request에서null 검사를 해주면 좋을 것 같아
속성 위에 @Notnull 어노테이션을 붙이고 API 파라미터 부분에 이것을 호출할 때 @Vaild를 붙이면 돼 

@Vaild에 대해서 한 번 공부해봐!

ghpark7 committed:
Update ProductApi.java

sunwon12 committed:
Feat: 공통 ResponseForm 구현 (#17)

sunwon12 committed:
Feat:  리뷰 삭제 API 구현 (#15)

* feat: 리뷰 삭제 API

* rename: 주석추가

GGHDMS committed:
Feature: 배송 관련 ERD 수정 (#20)

* fix: 맥북 m1 시리즈 오류 해결

* feat: 배송 erd 수정

하나의 주문에서 여러개의 배송이 발생할 수 있기 때문에
주문에 연결하는 것이 아닌 주문-상품에 연결해서
이를 해결함

sunwon12 committed:
Feat: aws s3 구축 및 저장 로직 구현 (#21)

* refactor: 시크릿 키 환경변수 설정으로 변경

* refactor: ResponseForm 도입에 따른 리팩토링

* feat: 미디어 파일 S3에 저장 관련 로직 구현

* refactor: 리뷰생성 시 리뷰이미지 저장 로직 추가

* refactor: fileName -> path 변수명 수정

GGHDMS committed:
Feat: 장바구니 테이블 추가하기 (#23)

* feat: 장바구니 Entity 추가

* fix:  빠진 import 추가

`@DeleteMapping` 의 import가 빠져 있어
실행시 빌드 오류가 발생해 추가함

* refactor: 불필요한 공백 제거

* refactor: 연관관계 변경

`@ManyToOne`이 맞는 표현이라 생각됨

* refactor: 잘못변경한 부분 수정과 원래 변경하려던 부분 수정

`@ManyToOne`이 맞는 표현이라 생각됨

ghpark7 committed:
Feat: Product 생성 API

ghpark7 committed:
refactor: url 수정

GGHDMS reviewed (approved):
👍
